Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -77,449 +66,452 @@ class RasterizeType(Enum):
+        self.offline_mode = offline_mode
+        self.tab = None
 
+        demisto.debug(f'TabLifecycleManager, __init__, {self.chrome_port=}, active tabs len: {len(self.browser.list_tab())}')
 
-def opt_name(opt):
-    return opt.split('=', 1)[0]
+    def __enter__(self):
+        self.tab = self.browser.new_tab()
 
+        self.tab.start()
+        if self.offline_mode:
+            self.tab.Network.emulateNetworkConditions(offline=True, latency=-1, downloadThroughput=-1, uploadThroughput=-1)
+        else:
+            self.tab.Network.emulateNetworkConditions(offline=False, latency=-1, downloadThroughput=-1, uploadThroughput=-1)
 
-def merge_options(default_options, user_options):
-    """merge the defualt options and user options
+        self.tab.Page.enable()
+        demisto.debug(f'TabLifecycleManager, {self.chrome_port=}, entering tab {self.tab.id},'
+                      f' tabs len: {len(self.browser.list_tab())}')
+        return self.tab
 
-    Arguments:
-        default_options {list} -- list of options to use
-        user_options {string} -- user configured options comma seperated (comma value can be escaped with \\)
+    def __exit__(self, exc_type, exc_val, exc_tb):  # pylint: disable=unused-argument
+        if exc_type or exc_val or exc_tb:
+            demisto.info(f'TabLifecycleManager, __exit__ with exception, {self.chrome_port=}, {exc_type=}, {exc_val=}, {exc_tb=}')
 
-    Returns:
-        list -- merged options
-    """
-    user_options = re.split(r'(?<!\\),', user_options) if user_options else []
-    if not user_options:  # nothing to do
-        return default_options.copy()
-    demisto.debug(f'user chrome options: {user_options}')
-    options = []
-    remove_opts = []
-    for opt in user_options:
-        opt = opt.strip()
-        if opt.startswith('[') and opt.endswith(']'):
-            remove_opts.append(opt[1:-1])
-        else:
-            options.append(opt.replace(r'\,', ','))
-    # remove values (such as in user-agent)
-    option_names = [opt_name(x) for x in options]
-    # add filtered defaults only if not in removed and we don't have it already
-    options.extend([x for x in default_options if (opt_name(x) not in remove_opts and opt_name(x) not in option_names)])
-    return options
+        if self.tab and self.tab.id:
+            tab_id = self.tab.id
+            threading.excepthook = excepthook_recv_loop
 
+            try:
+                time.sleep(1)  # pylint: disable=E9003
+                demisto.debug(f'TabLifecycleManager, __exit__, {self.chrome_port=}, disabling page')
+                self.tab.Page.disable()
+            except Exception as ex:
+                demisto.info(f'TabLifecycleManager, __exit__, {self.chrome_port=}, failed to disable page due to {ex}')
 
-def check_response(driver):
-    EMPTY_PAGE = '<html><head></head><body></body></html>'
-    if driver.page_source == EMPTY_PAGE:
-        return_err_or_warn(EMPTY_RESPONSE_ERROR_MSG)
+            try:
```

Comment:
Instead of using several try / catch for every few commands, please use 1 try catch and several except clauses based on the specific exception thrown. for example:
try:
... code
except Exception_A as ex_a:
... handle this case
except Exception_B as ex_b:
... handle ...

Reply:
Additionally, don't use broad catchers unless it's absolutely necessary

Reply:
Each try-except can have any Exceptions.
This is how I know in which step the exception occurred.